internal/metricsregistrar/create_metrics.go

```diff
@@ -0,0 +1,115 @@
+// Copyright 2021 the Exposure Notifications Server authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metricsregistrar
+
+import (
+	"context"
+	"fmt"
+	"runtime"
+	"time"
+
+	monitoring "cloud.google.com/go/monitoring/apiv3/v2"
+	"github.com/google/exposure-notifications-server/pkg/logging"
+	"github.com/google/exposure-notifications-server/pkg/observability"
+	"github.com/hashicorp/go-multierror"
+	"golang.org/x/sync/semaphore"
+	monitoringpb "google.golang.org/genproto/googleapis/monitoring/v3"
+)
+
+// createMetrics creates the upstream metrics in Stackdriver.
+func (s *Server) createMetrics(ctx context.Context) error {
+	logger := logging.FromContext(ctx)
+
+	logger.Infow("starting metric registration")
+	defer logger.Infow("finished metric registration")
+
+	// Extract the project ID.
+	sd := s.config.ObservabilityExporter.Stackdriver
+	if sd == nil {
+		return fmt.Errorf("observability export is not stackdriver")
+	}
+	projectID := sd.ProjectID
+
+	// Create the metrics client.
+	client, err := monitoring.NewMetricClient(context.Background())
+	if err != nil {
+		return fmt.Errorf("failed to create metrics client: %w", err)
+	}
+
+	// Create the Stackdriver exporter.
+	exporter, err := observability.NewStackdriver(context.Background(), &observability.StackdriverConfig{
+		ProjectID: projectID,
+	})
+	if err != nil {
+		return fmt.Errorf("failed to create Stackdriver exporter: %w", err)
+	}
+
+	// Register metric descriptors with Stackdriver.
+	allViews := observability.AllViews()
+
+	workers := int64(runtime.NumCPU())
+	if workers < 3 {
```

Comment:
Since there's exactly one instance of this, I felt like the parallelization was worth it.